src/execution_plan/optimizations/reduce_traversal.c

```diff
@@ -66,8 +66,8 @@ void reduceTraversal(ExecutionPlan *plan) {
 		   AlgebraicExpression_DiagonalOperand(ae, 0)) continue;
 
 		/* Search to see if dest is already resolved */
-		if(!ExecutionPlan_LocateOpResolvingAlias(op->children[0],
-												 AlgebraicExpression_Destination(ae))) continue;
+		if(!ExecutionPlan_LocateOpResolvingAliasExcludeApply(op->children[0],
```

Comment:
I hope we'll be able to reflect exactly which entities are "modified" by op_argument, thus avoiding this specific function.